scio-test/src/it/scala/com/spotify/scio/ScioContextIT.scala

```diff
@@ -54,4 +54,16 @@ class ScioContextIT extends FlatSpec with Matchers {
     ScioUtil.isRemoteUri(new URI(gcpTempLocation)) shouldBe true
   }
 
+  it should "#1323: generate unique SCollection names" in {
+    val options = PipelineOptionsFactory.create()
+    options.setRunner(classOf[DataflowRunner])
+    options.as(classOf[GcpOptions]).setProject(ItUtils.project)
+    val sc = ScioContext(options)
+
+    val s1 = sc.empty[(String, Int)]()
+    val s2 = sc.empty[(String, Double)]()
+    s1.join(s2)
+    sc.close()
+  }
```

Comment:
You should assert on something here?

Reply:
Not necessary. Beam throws an exception when the name is not unique which makes the test fail.

Reply:
There's a `noException should be thrownBy` matcher in scalatest, that could make this more explicit.